java/arcs/android/storage/service/SuspendingServiceMethods.kt

```diff
@@ -42,7 +44,12 @@ class ContinuationRegistrationCallback(
         }
 
         override fun onFailure(exception: ByteArray?) {
-            continuation.resumeWithException(Exception("Registration failed"))
+            val exceptionMessage = if (exception == null) {
+                "No exception provided"
+            } else {
+                decodeProto(exception, CrdtExceptionProto.getDefaultInstance()).message
+            }
+            continuation.resumeWithException(Exception("Registration failed: $exceptionMessage"))
```

Comment:
can we wrap the existing exception instead of just the message eg
val exception = decodeProto(exception, CrdtExceptionProto.getDefaultInstance())
continuation.resumeWithException(Exception("Registration failed", $exception))

Reply:
I started down that path, and didn't see an easy way to turn a `CrdtExceptionProto` back into an exception. Let me look a little harder.

Reply:
you are right we don't have a method for that, but I think we also want the stack trace that is in the proto. Otherwise you will only get "Exception occurred while registering callback" which is not very useful.
so maybe just concatenate the stack trace to the message?

Reply:
Maybe this makes the case for just printing them in BindingContext when they happen.